pxteditor/code-validation/getNestedChildBlocks.ts

```diff
@@ -0,0 +1,13 @@
+/// <reference path="../../localtypings/validatorPlan.d.ts" />
+
+import * as Blockly from "blockly";
+
+// returns the directly nested children of a block
+// for blocks with a mouth, it returns the blocks inside the mouth and its inputs, if any
+// for something like pick random, it would return the two number blocks
+export function getNestedChildBlocks(parentBlock: Blockly.Block): Blockly.Block[] {
+    const descendants = parentBlock.getDescendants(true);
+    const enabledDescendants = descendants.filter((block) => block.isEnabled());
+    const nestedChildren = enabledDescendants.filter((block) => block.getSurroundParent() === parentBlock);
```

Comment:
What you have is fine, but I think you could avoid the duplicate list iteration with:

`const nestedChildren = enabledDescendants.filter((block) => block.isEnabled() && block.getSurroundParent() === parentBlock);`